### PR TITLE
New allocator corrections

### PR DESCRIPTION
**Pull request by data-man**

### CODE REVIEW

**Comment by Snektron** [comment]
StackFallbackAllocator is also broken as of #5064. Perhaps it should be replaced by a generic fallback allocator which wraps
two allocators? (instead of hardcoding it to be FixedBufferAllocator.)

File: lib/std/heap/logging_allocator.zig [hunk 1]
```diff
@@ -1,57 +1,57 @@
 const std = @import("../std.zig");
 const Allocator = std.mem.Allocator;
 
-/// This allocator is used in front of another allocator and logs to the provided stream
-/// on every call to the allocator. Stream errors are ignored.
+/// This allocator is used in front of another allocator and logs to the provided writer
+/// on every call to the allocator. Writer errors are ignored.
 /// If https://github.com/ziglang/zig/issues/2586 is implemented, this API can be improved.
-pub fn LoggingAllocator(comptime OutStreamType: type) type {
+pub fn LoggingAllocator(comptime WriterType: type) type {
```
Comment (andrewrk) [inline lib/std/heap/logging_allocator.zig:7]: This should be changed to use `std.log` facilities, and it no longer needs to be a generic type.

Comment (marler8997) [reply lib/std/heap/logging_allocator.zig:7]: Does that mean we won't be able to test the logging output anymore?

Comment (andrewrk) [reply lib/std/heap/logging_allocator.zig:7]: we can add a "compare output" test case for it: https://github.com/ziglang/zig/blob/master/test/compare_output.zig

Comment (andrewrk) [reply lib/std/heap/logging_allocator.zig:7]: This gave me another idea:  #5738

Comment (data-man) [reply lib/std/heap/logging_allocator.zig:7]: @andrewrk 
> This should be changed to use std.log facilities

What to use?
`log.notice`, `log.info`, `log.debug`?

What scope? `.loggingAllocator`?

> we can add a "compare output" test case for it

How? With `log` function in `compare_output.zig`?

**Comment by andrewrk** [comment]
stale draft, please re-open when it's ready for review

**Pull request closed without merging**